juntagrico/decorators.py

```diff
@@ -23,17 +23,22 @@ def wrapper(request, *args, **kwargs):
     return wrapper
 
 
-def create_subscription_session(view):
-    @wraps(view)
-    def wrapper(request, *args, **kwargs):
-        som = SessionObjectManager(request, 'create_subscription', CSSessionObject)
-        session_object = som.data
-        if request.user.is_authenticated:
-            session_object.main_member = request.user.member
-        if session_object.main_member is None and request.resolver_match.url_name is not 'signup':
-            return redirect('signup')
-        response = view(request, som.data, *args, **kwargs)
-        som.store()
-        return response
-
-    return wrapper
+def create_subscription_session(view=None, *, session_object_class=CSSessionObject):
```

Comment:
do you need te session_object_class to be named parameter only?

Reply:
good points! :) If two apps needed to expand the sign-up process, the overwriting needed to take place in the "mother app". 

If I understand correctly, your idea to design juntagrico in a way that, if an additional app can add to the sign-up process automatically and independently of other apps, right?

We would then have a function of the style `addons.config.register_signup_step` for instance in our app, right?

I dont understand your last question "do you need te session_object_class to be named parameter only". Currently, the decorator could be called like a normal decorator without any named paramters, or with you named parameter called `session_object_class`